template/src/screens/DemoScreen.tsx

```diff
@@ -26,14 +29,31 @@ interface DemoScreenProps {
 }
 
 const DemoScreen = ({ navigation }: DemoScreenProps) => {
+  const [isLogoutLoading, setIsLogoutLoading] = useState(false)
   const counter = useAppSelector(selectCounter)
   const comicRequest = useAppSelector(selectComic)
   const dispatch = useAppDispatch()
   const { t } = useTranslation()
 
   useEffect(() => {
-    dispatch(getLatestComicAsync())
-  }, [])
+    if (isNotRequested(comicRequest)) {
+      dispatch(getLatestComicAsync())
+    }
+  }, [comicRequest.type])
+
+  const logOut = async () => {
+    try {
+      setIsLogoutLoading(true)
+      // typical logout for apps that require user to be logged in
+      // before giving any further access
+      persistor.pause()
+      await clearPersistence()
+      dispatch(resetStore())
+      persistor.persist()
```

Comment:
How does it work?

```
persistor.pause()
await clearPersistence()
dispatch(resetStore())
persistor.persist()
```

Reply:
it pauses redux-persist
it clears the persistence
it resets the store
it resumes redux-persist

a nicer way would be to use built-in redux-persist action `PURGE`, but it happened to be buggy